crates/ty_python_semantic/src/types.rs

```diff
@@ -6227,6 +6247,35 @@ impl<'db> InvalidTypeExpression<'db> {
     }
 }
 
+/// Data regarding a `warnings.deprecated` or `typing_extensions.deprecated` decorator.
+#[salsa::interned(debug)]
+#[derive(PartialOrd, Ord)]
+pub struct DeprecatedInstance<'db> {
+    /// The message for the deprecation
+    pub message: StringLiteralType<'db>,
+
+    /// The `@warnings.deprecated(...)` invocation
+    pub definition: Option<Definition<'db>>,
```

Comment:
```suggestion
    pub(crate) message: StringLiteralType<'db>,

    /// The `@warnings.deprecated(...)` invocation
    pub(crate) definition: Option<Definition<'db>>,
```

Is the `definition` used anywhere? If not, should we remove it or do you plan on using it in the diagnostic message?

Reply:
It's totally useless right now, I figured it might be useful in the future and was presumably cheap to include. I can just remove it though.